Java/LFU.java

```diff
@@ -0,0 +1,144 @@
+package lfu;
+import java.util.LinkedList;
+import java.util.ListIterator;
+import java.util.HashMap;
+
+/**
+ * @author salman
+ * @param <T>
+ */
+
+public class LFU<T> {
+
+	private int MAX_SIZE = 2;
+
+	private LinkedList< FrequencyList<T> > lfuList;
+	private HashMap<T, Pair<Integer, ListIterator<T> > > nodeFrequencyMap;
+	private HashMap<Integer, ListIterator< FrequencyList<T> > > frequencyMap;
+	private int count = 0;
+
+	public void init() {		
```

Comment:
If the sole function of this init() is to be called from the constructor, then place this code in constructor itself.